src/view/com/composer/text-input/TextInput.tsx

```diff
@@ -192,10 +194,32 @@ export const TextInput = forwardRef(function TextInputImpl(
     [onChangeText, richtext, setAutocompletePrefix],
   )
 
+  const splitter = useMemo(() => new Graphemer(), [])
   const textDecorated = useMemo(() => {
+    let excess
+    if (richtext.graphemeLength > MAX_GRAPHEME_LENGTH) {
+      const split = splitter.splitGraphemes(richtext.text)
+      const validText = split.slice(0, MAX_GRAPHEME_LENGTH).join('')
+      const validUnicode = new UnicodeString(validText)
+      const excessText = split.slice(MAX_GRAPHEME_LENGTH).join('')
+      richtext.delete(validUnicode.utf8.byteLength, richtext.length)
```

Comment:
This looks problematic, `useMemo` should be pure and should not mutate things passed into it. Any other ideas for how to structure this code? You can create a new rich text, modify it, and return it from here.

Reply:
Oh sure, no problem, `RichText` has a clone method which we can use here